en/sensor/vectornav.md

```diff
@@ -0,0 +1,119 @@
+1. Configure driver as either an external INS or to provide raw data:
+
+   - If using the VectorNav as an external INS, set [VN_MODE](../advanced_config/parameter_reference.md#VN_MODE) to `INS`. This disables EKF2.
+   - If using the VectorNav as external inertial sensors:
+
+     1. Set [VN_MODE](../advanced_config/parameter_reference.md#VN_MODE) to `Sensors Only`
+     1. If internal sensors are enabled, prioritize VectorNav sensors using [CAL_GYROn_PRIO](../advanced_config/parameter_reference.md#CAL_GYRO0_PRIO), [CAL_ACCELn_PRIO](../advanced_config/parameter_reference.md#CAL_ACCEL0_PRIO), [CAL_BAROn_PRIO](../advanced_config/parameter_reference.md#CAL_BARO0_PRIO), [CAL_MAGn_PRIO](../advanced_config/parameter_reference.md#CAL_MAG0_PRIO), where _n_ is the instance number of the IMU component (0, 1, etc.).
+
+     ::: tip
+     In most cases the extenal IMU (VN) is the highest-numbered.
+     You can get a list of the IMU components available using [`uorb top -1`](../middleware/uorb.md#uorb-top-command), you can differentiate between them using the [`listener`](../modules/modules_command.md#listener) command and looking through the data, or just the rates itself.
+
+     Alternatively, you can check [CAL_GYROn_ID](../advanced_config/parameter_reference.md#CAL_GYRO0_ID) to see the device id.
+     The priority is 0-255, where 0 is entirely disabled and 255 is highest priority.
+     :::
+
+1. Restart PX4
+
+Once enabled, the module will be detected on boot.
+IMU data should be published at 800Hz (400Hz if using VN-300).
+
+## VectorNav Configuration
+
+Upon initialization, PX4 configures the VectorNav unit as follows:
+
+- Enables necessary binary outputs
+- Disables ASCII outputs on active serial port
+- Configures VPE Heading Mode to Absolute
+- Autoscans baudrates and configures active port to 921600 bps
+
+All other necessary configuration parameters must be separately loaded to the VectorNav unit manually. These most commonly include:
+
+- `GNSS Antenna A Offset` Necessary if using a GNSS-enabled product and the GNSS antenna is mounted more than 10 cm from the VectorNav unit
+- `GNSS Antenna Baseline` Necessary if using a dual-GNSS-enabled product
+- `Reference Frame Rotation` Necessary if not mounted as connector-back, base-down
+- `IMU Filtering Config` Recommended to adjust the default 200Hz IMU filtering
+
+After setting these parameters, the settings must be set to persist over a power cycle using a Write Non Volatile command.
+
+## Published Data
+
+Upon initialization, the driver should print the following information to console (printed using `PX4_INFO`)
+
+- Unit model number
+- Unit hardware version
+- Unit serial number
+- Unit firmware number
+
+This should be accessible using the [`dmesg`](../modules/modules_system.md#dmesg) command.
+
+The VectorNav driver publishes the unit's data to the following UOrb topics:
+
+- [external_ins_local_position]()
+- [external_ins_global_position]()
+- [external_ins_attitude]()

```

Comment:
Can you provide links to these - not sure which UORB topics they are?

Reply:
@dagar 

I'm surprised not to see these messages in ~/PX4-Autopilot/msg, I thought that anything I can uorb (which I can call uorb external_ins_local_position) would be in that folder. Is there somewhere else it is aliased under, or do I have a misunderstanding?